### PR TITLE
Rewrite and simplify comment-format

### PR DESCRIPTION
**Pull request by ajafff**
#### PR checklist

- [ ] Addresses an existing issue: #0000
- [x] New feature, bugfix, or enhancement
  - [x] Includes tests
- [ ] Documentation update

#### Overview of change:
Initially I just wanted to convert the walker to a function. But then I needed to change so much it became a complete rewrite.

#### CHANGELOG.md entry:

[enhancement] `comment-format` no longer excludes comments with triple slash from linting except `/// <reference path="..."/>`

### CODE REVIEW

File: test/rules/comment-format/upper/test.ts.lint [hunk 1]
```diff
@@ -10,7 +10,8 @@ class Clazz { // This comment is correct
         console.log("test"); //This comment has no space
                                ~~~~~~~~~~~~~~~~~~~~~~~~~ [space]
     }
-    /// <reference or something>
+    /// <not a reference>
+      ~~~~~~~~~~~~~~~~~~~ [space]
```
Comment (ajafff) [inline test/rules/comment-format/upper/test.ts.lint]: Yeah, it should handle comments starting with more than 2 slashes better. The comment text should only be checked starting from the first non-slash character. 

And maybe there could be a new option to disallow more than 2 slashes

File: src/rules/commentFormatRule.ts [hunk 2]
```diff
@@ -103,136 +115,87 @@ export class Rule extends Lint.Rules.AbstractRule {
...
 
-class CommentWalker extends Lint.RuleWalker {
-    private exceptionsRegExp: ExceptionsRegExp;
-    private failureIgnorePart: string = "";
-
-    constructor(sourceFile: ts.SourceFile, options: Lint.IOptions) {
-        super(sourceFile, options);
+function parseOptions(options: Array<string | IExceptionsObject>): Options {
+    return {
+        case: options.indexOf(OPTION_LOWERCASE) !== -1
```
Comment (nchen63) [inline src/rules/commentFormatRule.ts]: nested ternary is discouraged

File: src/rules/commentFormatRule.ts [hunk 3]
```diff
@@ -103,136 +115,87 @@ export class Rule extends Lint.Rules.AbstractRule {
...
-                .join("|");
-
-            // regex is "start of string"//"any amount of whitespace"("any word from ignore list") followed by non alphanumeric character
-            return new RegExp(`^//\\s*(${wordsPattern})\\b`);
+        if (start === end) {
+            return;
+        }
+        const commentText = fullText.slice(start, end);
+        // whitelist //#region and //#endregion and JetBrains IDEs' "//noinspection ..."
+        if (/^(?:#(!?end)?region|noinspection\s)/.test(commentText)) {
```
Comment (nchen63) [inline src/rules/commentFormatRule.ts]: what is the `!?end` part for? This matches `!end`

Comment (ajafff) [inline src/rules/commentFormatRule.ts]: You're right, that should be `?:end`

**Pull request merged**